packages/core/react/src/__tests__/WalletProviderDesktop-test.tsx

```diff
@@ -0,0 +1,240 @@
+/**
+ * @jest-environment jsdom
+ */
+
+'use strict';
+
+import 'jest-localstorage-mock';
+
+import type { Adapter, WalletName } from '@solana/wallet-adapter-base';
+import { BaseWalletAdapter, WalletReadyState } from '@solana/wallet-adapter-base';
+import React, { createRef, forwardRef, useImperativeHandle } from 'react';
+
+import { PublicKey } from '@solana/web3.js';
+import type { WalletContextState } from '../useWallet.js';
+import { WalletProvider } from '../WalletProvider.js';
+import type { WalletProviderProps } from '../WalletProvider.js';
+import { act } from 'react-dom/test-utils';
+import { createRoot } from 'react-dom/client';
+import { useWallet } from '../useWallet.js';
+
+jest.mock('../getEnvironment.js', () => ({
+    ...jest.requireActual('../getEnvironment.js'),
+    __esModule: true,
+    default: () => jest.requireActual('../getEnvironment.js').Environment.DESKTOP_WEB,
```

Comment:
Other than mocking `getEnvironment` differently, these two tests are identical. They will start to diverge in future commits, as mobile functionality is introduced.